File: Cpp/1512_Number_of_Good_Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int result = 0;
        unordered_map<int, int> m;
        
        for(int i = 0; i < nums.size(); ++i) {
            ++m[nums[i]];
        }
        
        for(auto& it : m) {
            result += (it.second * (it.second-1)) / 2;
        }
        
        return result;
    }
};

File: Kotlin/1512_Number_of_Good_Pairs.kt
class Solution {
    fun numIdenticalPairs(nums: IntArray): Int {
        var result = 0
        val m = mutableMapOf<Int, Int>()
        
        nums.forEach {
            m.put(it, m.getOrDefault(it, 0) + 1)
        }

        m.forEach { (key, value) ->
            result += (value * (value - 1)) / 2
        }
        
        return result
    }
}